google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,251 @@
+
+    def add_rule(self, rule_def, rule_index):
+        """Add a rule to the rule book.
+
+        Add a rule to the rule book.
+
+        The rule supplied to this method is the dictionary parsed from
+        the rules definition file.
+
+        For example, this rule...
+
+            # rules yaml:
+            rules:
+              - project: a rule
+                network: whitelist
+                is_externam
+                whitelist:
+                  - project:network
+
+        ... gets parsed into:
+
+            {
+                'name': 'a rule',
+                'resource': {
+                    'project': '',
+                    'network': self,
+                    'is_external': boolean
+                }
+                whitelist {
+                    'network': ['project'...]
+
+                }
+            }
+
+        Args:
+            rule_def: A dictionary containing rule definition properties.
+            rule_index: The index of the rule from the rule definitions.
+                Assigned automatically when the rule book is built.
+
+        Args:
+            rule_def: A dictionary containing rule definition properties.
+            rule_index: The index of the rule from the rule definitions.
+            Assigned automatically when the rule book is built.
+        """
+        project = rule_def.get('project')
+        network = rule_def.get('network')
+        whitelist = rule_def.get('whitelist')
+        is_external_network = rule_def.get('is_external_network')
+
+        if (whitelist is None) or (project is None) or (network is None)\
+                or (is_external_network is None):
+            raise audit_errors.InvalidRulesSchemaError(
+                'Faulty rule {}'.format(rule_def.get('name')))
+
+        rule_def_resource = {'whitelist': whitelist,
+                             'project': escape_and_globify(project),
+                             'network': escape_and_globify(network),
+                             'is_external_network': is_external_network}
+
+        rule = Rule(rule_name=rule_def.get('name'),
+                    rule_index=rule_index,
+                    rules=rule_def_resource)
+
+        resource_rules = self.resource_rules_map.get(rule_index)
+        if not resource_rules:
+            self.resource_rules_map[rule_index] = rule
+
+    def get_resource_rules(self):
+        """Get all the resource rules for (resource, RuleAppliesTo.*).
+
+        Returns:
+            A list of ResourceRules.
+        """
+        return self.resource_rules_map.values()
+
+
+class Rule(object):
+    """The rules class for instance_network_interface."""
+
+    def __init__(self, rule_name, rule_index, rules):
+        """Initialize.
+
+        Args:
+            rule_name: Name of the loaded rule
+            rule_index: The index of the rule from the  definitions
+            rules: the network instance
+            the resources to accosiate with the rules
```

Comment:
Could you indent the continuation lines, e.g.

```
    rules: The resource to associate 
        with the rules (for now, every network_instance)
```

I'm also not clear on the `rules` arg description, if you could clarify, that would be great. Also, super nit... please capitalize the first word of the phrase.